akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/FileUploadExamplesSpec.scala

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package docs.http.scaladsl.server
+
+import java.io.File
+
+import akka.http.scaladsl.model.Multipart.FormData.BodyPart
+import akka.stream.io.SynchronousFileSink
+import akka.stream.scaladsl._
+import akka.http.scaladsl.model.Multipart
+
+import scala.concurrent.Future
+
+class FileUploadExamplesSpec extends RoutingSpec {
+
+  //#simple-upload
+  val simpleRoute =
+    path("upload") {
+      entity(as[Multipart.FormData]) { formData =>
+        val completed: Source[File, _] = formData.parts.mapAsync(1) {
+          case BodyPart(name, entity, _, _) =>
+            val file = File.createTempFile("uploaded", "bin")
+            val sink = SynchronousFileSink(file)
+            entity.dataBytes.runWith(sink).map(_ => file)
```

Comment:
also, `BodyBart("not-a-file", ) =>`
so we can for example get metadata about a movie...

Maybe that's good for the low-level case.

3 examples:
- low level, multiple bory parts - one metadata, one file
- simple file upload to file, as shown below
- streaming handling - split by line, handle some csv